### PR TITLE
👌 IMPROVE: Compute file size, when writing to zip archive

### PR DESCRIPTION
**Pull request by chrisjsewell**
This allows for zipfile to decide if it should use the ZIP64 extension (when adding files >2Gb).

Note `ZipPath.putfile` does this internally

Should fix: https://github.com/aiidateam/aiida-core/discussions/5407#discussioncomment-2599275

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Not sure how to test this, because I guess it would involve writing a >2 Gb file?

**Comment by chrisjsewell** [comment]
this also is currently awaiting a new release on https://github.com/conda-forge/archive-path-feedstock

**Comment by chrisjsewell** [comment]
> this also is currently awaiting a new release on [conda-forge/archive-path-feedstock](https://github.com/conda-forge/archive-path-feedstock)

release done, but it'll take a while to show up for conda install

File: aiida/tools/archive/implementations/sqlite_zip/writer.py [hunk 1]
```diff
@@ -164,7 +165,13 @@ def _stream_binary(
         if compression is not None:
             kwargs['compression'] = zipfile.ZIP_DEFLATED if compression else zipfile.ZIP_STORED
             kwargs['level'] = compression
-        with self._zip_path.joinpath(name).open('wb', **kwargs) as zip_handle:
+
+        # compute the file size of the handle
+        handle.seek(0, os.SEEK_END)
+        file_size = handle.tell()
+        handle.seek(0)
```
Comment (sphuber) [inline aiida/tools/archive/implementations/sqlite_zip/writer.py:172]: I guess in most cases the position of the handle will be `0` here, but it is not guaranteed right? To be absolutely safe, you might want to do the following:
```suggestion
        position = handle.tell()
        handle.seek(0, os.SEEK_END)
        file_size = handle.tell()
        handle.seek(position)
```

**Review by sphuber — COMMENTED** [review]
Cheers, just one comment. As for testing, a clean solution is indeed not easy. I think if we rerun @mbercx 's failing example with this branch and it works, that should be sufficient for now.

**Pull request merged**